SGoopas/Assets/Scripts/Game/Controller/ControllerConfigure.cs

```diff
@@ -52,6 +59,9 @@ private void ConfigureControls()
         playerStateMachine = new MasterPlayerStateMachine(MainObjectContainer.Instance.Player2D, MainObjectContainer.Instance.Player3D);
         ConfigureControls();
 
+        DimensionSwitchHandler.ResetDimensionSwitchEvent();
```

Comment:
i see, so this is the event that triggers the message event, which will be called by ShadowSwapIntro. Dude this event based programming is really handy